### PR TITLE
Added support for several mods

### PR DESCRIPTION
**Pull request by vizthex123**
- Bigger Reactors (also renamed the file for Extreme Reactors to make it clearer)
- Deeper Caves
- Deep Resonance
- Galosphere (added Salt Blocks since Silver Ore was already done)
- The Outer End
- Project Red
- Regions Unexplored
- RF Tools
- Thermal Series
- XyCraft: World (only Kivi and Aluminum since Xychorium ores don't copy over textures)
- Yung's Cave Biomes

### CODE REVIEW

**Comment by vizthex123** [comment]
I'm hoping the mod dev can figure out why Xychorium variants don't have textures (mentioned it #205). I think it uses a custom shader or something that Dynamic Asset Generator isn't properly detecting.

**Comment by lukebemish** [comment]
I'll look into the missing textures; I suspect you're write, they probably have some sort of custom model thingy I can't detect. I'll try to review the rest of this PR when I have a chance, though 1.20.1 is fairly low priority right now so it may be a bit.

**Comment by vizthex123** [comment]
Alright, that sounds good.

Just wanted to toss in what I'd done to save you some time (I figured it probably wasn't too hard to release an update for 1.20.1 if code wasn't changed) and support a couple major-*ish* mods that were missing it.

**Comment by lukebemish** [comment]
Sorry I've been so delayed in looking at this! I should have more free time soon so I'll plan on getting it looked over and merged soon-ish.

**Comment by vizthex123** [comment]
noice, thanks.

~~If you're able to check why it's making Slate variations of vanilla ores when Deeper Caves is installed it'd be nice. The mod adds them already, and I tried to blacklist the default ores but it's not working for me~~ - finally fixed that in https://github.com/lukebemishprojects/ExcavatedVariants/pull/207/commits/bd4d1cb83cd289bc5fcea0f91cf109e15a576e10

If you could check on Xychorium ores not having textures it'd be great. I can open another PR to add them, or add them to this one before it gets merged.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:59]: I feel like these ought to have at least one type, to actually generate any variants?

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5]: Uh, what happened to this file?

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5]: Stone IDs should be prefixed in case two mods add a stone with the same name -- `cavesanddepths_chert`, for instance. The same goes for stuff throughout the PR.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/galosphere.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/galosphere.json5]: Same as above w.r.t stone IDs

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/outer_end.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/outer_end.json5]: Same as above w.r.t stone IDs

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/regions_unexplored.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/regions_unexplored.json5]: Same as above w.r.t stone IDs

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/xycraft.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/xycraft.json5]: Same as above w.r.t stone IDs

**Review by lukebemish — CHANGES_REQUESTED** [review]

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:59]: No, the point is to blacklist it. They're just reskins of vanilla ores, and I didn't want the mod to generate duplicate Slate versions of each one since it's already covered by vanilla ores existing.

I think I told you about it in my issue report?

Either way though, if they're not blacklisted Excavated Variants makes a duplicate version of Slate for all vanilla ores - which is unnecessary & annoying since it already does that for vanilla ores.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5]: I don't even remember lol. I think I re-organized it and accidently copied it over?

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5]: Alright, I guess I can change it.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (lukebemish) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:59]: What, no. It _won't_ generate duplicate versions if you use the same ore ID, cause the entries from this and the entries from the vanilla coal ore file will get merged. The issue is that you're using the ID `slate_coal_ore` for some reason so it thinks its another ore, cause, well, you _told_ it that it is.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5
Comment (lukebemish) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5]: This PR shouldn't touch this file if you're not changing something in it

Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/minecraft.json5]: Yeah, I think I removed the spacing from the multi-line comment or something. It's been months, so I don't remember - but since it's showing no changes, it would've been something minor like that.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5:60]: Can we make sure files have trailing newlines?

**Review by lukebemish — CHANGES_REQUESTED** [review]

**Comment by lukebemish** [comment]
Once the remaining feedback is addressed I'll get this merged. I can handle porting it forward to newer versions myself; that shouldn't be too bad. Thank you a bunch for contributing!

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5:60]: What do you mean? Empty spaces at the end?

I guess I could, but I'm not sure why it should be there (and it seems like other files don't have them).

Comment (lukebemish) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5:60]: Eh, you're right, looks like I didn't fix that issue till a newer version, so no need. Just fix the other stuff and it's good

Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepresonance.json5:60]: Fixed them all.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:60]: The problem is that doing so makes it entirely remove the Slate Ores. That's why I set up this blacklist system. It's the only way to get the mod to keep the Slate Ores while not adding duplicates for them.
image

Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/deepercaves.json5:59]: Had issues with it before (that's why I made the workaround), but it seems to be working now. Finally lol.

**Comment by vizthex123** [comment]
Got everything resolved, and yeah I don't mind. If I ever find more mods missing support, I'll open another PR.

If you could check on Xychorium and add it in an update alongside this PR it'd be amazing. Would love to give all XyCraft ores support for this mod.

**Review by lukebemish — CHANGES_REQUESTED** [review]
Still need to fix whatever bunch of random changes happened to that one file...

**Comment by vizthex123** [comment]
re-downloaded it off the github.

**Review by lukebemish — APPROVED** [review]

**Comment by lukebemish** [comment]
Looks good to me; I'll get it merged when I've got some free time

**Comment by vizthex123** [comment]
Alrighty.

I'll add support for Project Red too.

**Comment by lukebemish** [comment]
Please do not modify this PR further before I merge it. I'll have to review any modifications you make and that'll set it back. Don't feel bad about splitting stuff into separate PRs

**Comment by vizthex123** [comment]
Sorry, already did that before you commented (or at least before I saw you left a comment). Still not used to making a ton of small commits & whatnot (complete opposite of how I do things lol, and I *really* hate having to make a ton of tiny changes).

I'll keep it in mind for next time though.

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/projectred.json5
Comment (lukebemish) [inline Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/projectred.json5]: Slightly confused about this -- is there supposed to be a thermal ore in this file?

**Review by lukebemish — CHANGES_REQUESTED** [review]

File: Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/projectred.json5
Comment (vizthex123) [reply Common/src/main/resources/defaultresources/excavated_variants/excavated_variants/variants/projectred.json5]: No, I did it by accident and fixed it in the next commit (which I forgot to push until now lol). Copied the Thermal file as a base for this one since they share a few ores.

**Review by lukebemish — APPROVED** [review]
Looks good. I'll get it out in the next release -- may be just a bit as I have to get all the stuff for that older version working again and port stuff forwards before the release.

**Comment by vizthex123** [comment]
noice, thanks. Looking forward to it.

**Comment by lukebemish** [comment]
That last commit was (intentionally) reverted and not included when I merged the PR -- if you want to submit it, it should be a separate PR (in fact compat for different mods should be in general) -- adding more stuff after a PR is approved just means that it has to go through the approval process again. The other changes will be released for 1.20.1 as soon as I've got it ported to newer versions (1.21.x) as well.

**Comment by vizthex123** [comment]
yeah, i couldn't figure out how to open another one when one was already open (i'm really dumb and only use github for backups lol).

but i'll see if i can do it now.

thanks for merging it.

**Pull request merged**
